### PR TITLE
Added gas fee strategy for remote tests.

### LINKED ISSUE #1213: Txs to polygon hang. Why: gas price it tried << gas price needed
### Background

I'm working with polygon network to grab predict-eth round 2 results. I'm in the step where I'm given a datatoken address, and I need to call `ocean.assets.download_asset()`. Polygon block latency is a few seconds. However it takes me 2-5 min for a tx to go through. 

I observed that the gas price I'm giving (31.9058) is much lower than what polygon matic gas station suggests. That's the problem! (Or at least one of them). Details below.

### TODOs for this issue

Ensure that we pay a more reasonable gas price in Polygon, so that users aren't waiting around. (Without reinventing the wheel. Brownie will likely make this easy, if we're smart about it.)

### Details of my call

This is in `predict-eth-judges` repo.

```text
(venv) trentmc@trentmc-xps:~/code/pej$ pej getpreds outdir-round2/
pej getpreds: Begin
Args: CSV_DIR=outdir-round2/
alice_wallet.address=0xA54ABd42b11B7C97538CAD7C6A2820419ddF703E. bal=9.928745801526913 MATIC
Loaded 2 DT addrs.
DT #1/2: Begin
dt_addr=0xb8187192f79E88eF942E50Ba433B71b6d226D5e1, nft_addr=0x53105b05E5a78B26e70aE6E6169C7e9CDe15b47A, did=did:op:fe8ffa177b719476bef1b8cbfb28b4e8043515817304812f018554afd04d7ef0
INFO:ocean_lib.data_provider.data_service_provider:Service initialized successfully initializeEndpoint https://v4.provider.polygon.oceanprotocol.com/api/services/initialize
Transaction sent: 0x2cf67f1596a71bd47f0e61ec7242a6dd9b2ed09324863c61f4d5b567dc59564a
  Gas price: 31.905822902 gwei   Gas limit: 96654   Nonce: 8
```

The key thing to note here is "Gas price: 31.905822902 gwei".

### Details of gas station

And, doing a call (as of Dec 14, 21:00 UCT) to https://gasstation-mainnet.matic.network/v2 outputs the following: `{"safeLow":{"maxPriorityFee":35.848027238,"maxFee":44.510349484},"standard":{"maxPriorityFee":36.741326008,"maxFee":45.403648254000004},"fast":{"maxPriorityFee":39.64206791064286,"maxFee":48.30439015664286},"estimatedBaseFee":8.662322246,"blockTime":3,"blockNumber":36834265}`

Note that even the lowest of these values is _much_ higher than the gas price that our tx gave.

### Details: my current brownie network-config.yaml

```text
development:
  - name: development
    chainid: 8996
    host: http://127.0.0.1:8545
    id: development
    required_confs: 1
live:
  - name: polygon-test
    networks:
      - chainid: 80001
        host: https://rpc-mumbai.maticvigil.com
        id: polygon-test
        required_confs: 1
  - name: polygon-main
    networks:
      - chainid: 137
        host: https://polygon-rpc.com
        id: polygon-main
        required_confs: 15
```

### PR DESCRIPTION
**Pull request by mariacarmina**
Fixes #1213  .

Changes proposed in this PR:

- querying gas station from polygon
- updated tx dicts
- deleted util function for setting the priority fee

### CODE REVIEW

File: tests/integration/remote/util.py [hunk 1]
```diff
@@ -105,8 +107,15 @@ def do_ocean_tx_and_handle_gotchas(ocean, alice_wallet):
 
     print("Call create() from data NFT, and wait for it to complete...")
     try:
+        # Polygon & Mumbai uses EIP-1559. So, dynamically determine priority fee
+        gas_fees = requests.get("https://gasstation-mainnet.matic.network/v2").json()
```
Comment (calina-c) [inline tests/integration/remote/util.py:111]: there is some duplication with the non_ocean_tx. Could we make a function that does the request and returns the priority fee and the max fee?

Could we also add some fallbacks, i.e. if the response from the API is not 200 or any key doesn't exist, we should handle that (maybe fail the tests gracefully?)

**Review by calina-c — CHANGES_REQUESTED** [review]
Good work, but it needs a bit more DRY

File: tests/integration/remote/util.py (hunk 1 above)
Comment (mariacarmina) [reply tests/integration/remote/util.py]: Ok, I will also add a fallback for that response.

File: tests/integration/remote/util.py [hunk 2]
```diff
@@ -41,6 +43,23 @@ def remote_config_polygon(tmp_path):
...
+    gas_resp = requests.get("https://gasstation-mainnet.matic.network/v2")
+
+    if not gas_resp or gas_resp.status_code != 200:
+        print(
+            f"Invalid response from Polygon gas station. Retry with brownie values..."
+        )
+
+        return chain.priority_fee, chain.base_fee + 2 * chain.priority_fee
+
+    return Web3.toWei(gas_resp.json()["fast"]["maxPriorityFee"], "gwei"), Web3.toWei(
```
Comment (calina-c) [inline tests/integration/remote/util.py:58]: nice work. If I were to nitpick, I would recommend assigning gas_resp.json() e.g. `gas_resp = gas_resp.json()` before, to make this easier to read, but only if you want to. Also, there's an empty line in nonocean_tx after the try which I would remove, but that is also minor. I'm approving, feel free to merge now or with these tiny changes.

**Review by calina-c — APPROVED** [review]

**Pull request merged**